Common/IsolatorLimitResultProvider.cs

```diff
@@ -0,0 +1,110 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using QuantConnect.Scheduling;
+
+namespace QuantConnect
+{
+    /// <summary>
+    /// Provides access to the <see cref="NullIsolatorLimitResultProvider"/> and extension methods supporting <see cref="ScheduledEvent"/>
+    /// </summary>
+    public static class IsolatorLimitResultProvider
+    {
+        /// <summary>
+        /// Provides access to a null implementation of <see cref="IIsolatorLimitResultProvider"/>
+        /// </summary>
+        public static readonly IIsolatorLimitResultProvider Null = new NullIsolatorLimitResultProvider();
+
+        /// <summary>
+        /// Convenience method for invoking a scheduled event's Scan method inside the <see cref="IsolatorLimitResultProvider"/>
+        /// </summary>
+        public static void Consume(
+            this IIsolatorLimitResultProvider isolatorLimitProvider,
+            ScheduledEvent scheduledEvent,
+            DateTime scanTimeUtc
+            )
+        {
+            // perform initial filtering to prevent starting a task when not necessary
+            if (scheduledEvent.NextEventUtcTime > scanTimeUtc)
```

Comment:
Isn't the caller already checking this?

Reply:
There's more than one caller and in `BacktestingRealTimeResultHandler` there's two difference guard clauses around the invocation. They actually have different intent as well. Also, in general I prefer to not make decisions in a method based on it's consumers. Now, the inverse isn't true -- so if there's a change that _should_ be made here (I'm not saying there is), then I would want to change the consumer of this function and not this function. The purposes of this guard clause is to prevent spinning up the task in the function in invokes just so the scheduled event can say 'its not time yet'.